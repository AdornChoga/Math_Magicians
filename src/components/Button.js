import PropTypes from 'prop-types';

const Button = (props) => {
  const { buttonValue } = props;
  const { eventHandler } = props;
  return (
    <li>
      <button type="button" onClick={eventHandler}>{ buttonValue }</button>
    </li>
  );
};

Button.propTypes = {
  buttonValue: PropTypes.string.isRequired, eventHandler: PropTypes.func.isRequired,
};
export default Button;
