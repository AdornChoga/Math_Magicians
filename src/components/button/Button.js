import PropTypes from 'prop-types';

const Button = (props) => {
  const { buttonValue, eventHandler, id } = props;
  return (
    <li>
      <button type="button" onClick={eventHandler} data-testid={id}>{ buttonValue }</button>
    </li>
  );
};

Button.propTypes = {
  buttonValue: PropTypes.string.isRequired,
  eventHandler: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
};
export default Button;
