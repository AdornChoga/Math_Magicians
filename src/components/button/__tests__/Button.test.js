import {
  render, screen, cleanup, fireEvent,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button from '../Button';

afterEach(cleanup);
describe('Tests button rendering and snapshot matching', () => {
  test('button should contain passed value', () => {
    const handleClicks = () => (5 + 3);
    render(<Button buttonValue="-" eventHandler={handleClicks} id="minus" />);
    const btnElement = screen.getByTestId('minus');
    expect(btnElement).toBeInTheDocument();
    expect(btnElement).toHaveTextContent('-');
  });

  test('is button snapshot created', () => {
    const handleClicks = () => (5 + 2);
    const tree = renderer.create(<Button buttonValue="-" eventHandler={handleClicks} id="minus" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Tests for click events on button', () => {
  test('is click event captured', (done) => {
    const handleClicks = (evt) => {
      expect(evt.target.value).toBe('-');
      done();
    };
    const { getByText } = render(<Button buttonValue="-" eventHandler={handleClicks} id="minus" />);
    const node = getByText('-');
    fireEvent.click(node, { target: { value: '-' } });
  });
});
