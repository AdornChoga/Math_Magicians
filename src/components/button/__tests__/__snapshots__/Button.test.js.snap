// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests button rendering and snapshot matching is button snapshot created 1`] = `
<li>
  <button
    data-testid="minus"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
</li>
`;
